google/cloud/forseti/common/util/regular_exp.py

```diff
@@ -39,3 +39,19 @@ def escape_and_globify(pattern_string):
     if pattern_string == '*':
         return '^.*$'
     return '^{}$'.format(re.escape(pattern_string).replace('\\*', '.+?'))
+
+
+def all_match(pattern_to_val):
+    """Lazily determines whether the given regexes match the values.
```

Comment:
In the previous list comprehension method, all regex matches would be evaluated into a list and then all() would be called on that list. Now, as soon as the first regex fails to match it will return (so the rest don't get evaluated). I am not sure if lazy is the right word but the idea was it lazily evaluates each regex.. i.e. doesn't do it if not needed. Let me know if you prefer another description. If you don't care too much about this micro optimization I can put the list comprehension back. Doubt it will be any difference in speed in reality.

Reply:
I was thinking lazy evaluation is more on a line of not creating all the values at the same time (maybe doing so through a generator). I am not concerned about the performance in this case, just a bit confused about the wording in the docstring.